ee/vellum_cli/tests/test_pull.py

```diff
@@ -822,3 +822,44 @@ def test_pull__multiple_instances_of_same_module__keep_when_pulling_another_modu
     with open(lock_json) as f:
         lock_data = json.load(f)
         assert len(lock_data["workflows"]) == 3
+
+
+def test_pull__module_name_from_deployment_name(vellum_client):
+    # GIVEN a workflow deployment
+    workflow_deployment = "test-workflow-deployment-id"
+
+    # AND the workflow pull API call returns a zip file with metadata containing a deployment_name
+    deployment_name = "Test Deployment"
+    vellum_client.workflows.pull.return_value = iter(
+        [
+            _zip_file_map(
+                {"workflow.py": "print('hello')", "metadata.json": json.dumps({"deployment_name": deployment_name})}
```

Comment:
[blocking] this test will fail if we added a `"label": "Some Other Label"` to this json